### PR TITLE
Bug 1823143: Add ICSP option to oc adm release extract

### PR DESCRIPTION
**Pull request by zaneb**

### CODE REVIEW

**Comment by zaneb** [comment]
Not an expert, but the test failures don't _look_ related.
/retest

**Review by soltysh — COMMENTED** [review]
A couple minor nits, I still need to test this one out, after I get mirror completed.

File: pkg/cli/image/extract/extract.go [hunk 1]
```diff
@@ -301,6 +304,10 @@ func parseMappings(images, paths, files []string, requireEmpty bool) ([]Mapping,
 		if len(src.Ref.Tag) == 0 && len(src.Ref.ID) == 0 {
 			return nil, fmt.Errorf("source image must point to an image ID or image tag")
 		}
+		if !icspWarned && hasICSPFile && len(src.Ref.Tag) > 0 {
+			fmt.Fprintf(errOut, "warning: --icsp-file only applies to images referenced by digest and will be ignored for tags\n")
```
[the remark this answers is not in the document]
Comment (soltysh) [reply pkg/cli/image/extract/extract.go:308]: ICSP is not meant for tags, not the current implementation. The latter version will have that extension.

**Comment by soltysh** [comment]
Superseded with https://github.com/openshift/oc/pull/1169
/close

**Pull request closed without merging**